### PR TITLE
Implement IdentifyMissingTests tool

### LINKED ISSUE #1652: Create a tool to tell us what tests we are missing
Let's create a console tool to grab the attributes created on #1655 and create a report.
This could be a good start https://github.com/microsoft/playwright-sharp/blob/main/src/tools/PlaywrightSharp.Tooling/IdentifyMissingTests.cs

### LINKED ISSUE #1654: Create a project with an attribute to identify tests
Let's mimic pw# https://github.com/microsoft/playwright-sharp/tree/main/src/tests/PlaywrightSharp.Xunit

### PR DESCRIPTION
**Pull request by Androbin**
Closes #1652
Closes #1654 @kblok

### CODE REVIEW

**Comment by kblok** [comment]
Conflicted @Androbin

File: lib/tools/PuppeteerSharp.Tooling/ScaffoldTest.cs [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+namespace PuppeteerSharp.Tooling
+{
+    [SuppressMessage("StyleCop.CSharp.ReadabilityRules", "SA1118:ParameterMustNotSpanMultipleLines", Justification = "CodeDom is complicated.")]
+    internal static partial class ScaffoldTest
+    {
+        private static readonly TextInfo _textInfo = CultureInfo.InvariantCulture.TextInfo;
+
+        public static void FindTestsInFile(string path, Action<string> callback)
+        {
+            var rx = new Regex(@"it\(\'(.*)\',");
```
Comment (kblok) [inline lib/tools/PuppeteerSharp.Tooling/ScaffoldTest.cs:44]: We should also consider `itFailsFirefox` here

**Comment by Androbin** [comment]
@kblok I ended up taking inspiration from https://github.com/hardkoded/puppeteer-sharp/issues/1652#issuecomment-803219383 after all

**Comment by Androbin** [comment]
Test run:
```
Total matching tests: 6/16.
Total tests found by name, but not by file: 0/16.
Total missing tests: 10/16.
accessibility.spec.ts: should work
accessibility.spec.ts: should report uninteresting nodes
accessibility.spec.ts: roledescription
accessibility.spec.ts: orientation
accessibility.spec.ts: autocomplete
accessibility.spec.ts: multiselectable
accessibility.spec.ts: keyshortcuts
accessibility.spec.ts: non editable textbox with role and tabIndex and label should not have children
accessibility.spec.ts: checkbox with and tabIndex and label should not have children
accessibility.spec.ts: checkbox without label should not have children
Found/Mismatched/Missing: 6/0/10 out of 16
```

**Comment by kblok** [comment]
@Androbin I think the numbers should be way way higher. Around the 600s.

**Comment by Androbin** [comment]
@kblok `ScaffoldTest.FindTestsInFile` finds the Puppeteer tests but `IdentifyMissingTests` doesn't recognize them as missing. I took the code straight from PlaywrightSharp and it seems to be doing it wrong.

**Comment by Androbin** [comment]
```
Total matching tests: 16/745.
Total tests found by name, but not by file: 0/745.
Total missing tests: 729/745.
...
accessibility.spec.ts: root option: should work a button
accessibility.spec.ts: root option: should work an input
accessibility.spec.ts: root option: should work a menu
accessibility.spec.ts: root option: should return null when the element is no longer in DOM
accessibility.spec.ts: root option: should support the interestingOnly option
...
defaultbrowsercontext.spec.ts: DefaultBrowserContext: page.cookies() should work
defaultbrowsercontext.spec.ts: DefaultBrowserContext: page.setCookie() should work
defaultbrowsercontext.spec.ts: DefaultBrowserContext: page.deleteCookie() should work
fixtures.spec.ts: Fixtures: dumpio option should work with pipe option 
fixtures.spec.ts: Fixtures: should dump browser process stderr
fixtures.spec.ts: Fixtures: should close the browser when the node process closes
Found/Mismatched/Missing: 16/0/729 out of 745
```

**Comment by kblok** [comment]
Good Good!

**Review by kblok — APPROVED** [review]

**Comment by kblok** [comment]
We need to add the strong name props to PuppeteerSharp.Xunit @Androbin

**Comment by Androbin** [comment]
CI errors are due to rules in newly added `stylecop.json`
@kblok I suggest we disable them for now, and create a ticket for fixing and re-enabling them

**Comment by Androbin** [comment]
@kblok In `SignAssembly.props`, how do I conditionally change the path from `..\keys\public.snk` to `..\..\keys\public.snk`?

**Comment by kblok** [comment]
What do you think about moving the new projects to the root level?

**Comment by Androbin** [comment]
Sure, why was it done this way in PlaywrightSharp though?

**Comment by kblok** [comment]
because we have another tool there. But we are going to remove it.

**Comment by Androbin** [comment]
@kblok Ready when you are

**Pull request merged**